src/main.rs

```diff
@@ -1,32 +1,63 @@
-use crate::trackpoints::places::SavedPlace;
+#[macro_use]
+extern crate diesel;
+
+//use crate::trackpoints::places::SavedPlace;
+//use std::path::PathBuf;
 use structopt::StructOpt;
 use walkdir::WalkDir;
 
 mod trackpoints;
+mod google;
+mod db;
 
 #[derive(Debug, StructOpt)]
 #[structopt(name = "excavator")]
 struct Opt {
     command: String,
-    file_name: String,
+    directory_name: String,
     //#[structopt(parse(from_os_str))]
     //file: PathBuf,
 }
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     let args = Opt::from_args();
-    let file_name = &args.file_name;
 
-    for entry in WalkDir::new(".")
+    let directory_name = &args.directory_name;
+
+    for entry in WalkDir::new(directory_name)
         .follow_links(true)
         .into_iter()
         .filter_map(|e| e.ok())
     {
         let f_name = entry.file_name().to_string_lossy();
-        if f_name.starts_with(file_name) {
+        let d_name = entry.path().to_string_lossy();
+        
+        if f_name.starts_with( "MyActivity.json") {
```

Comment:
Great work! Using pattern syntax with `f_name` here will be more clear and readable, I think.
Or I can handle that in another branch, thanks.